### ISSUE TITLE
Parsing of non-orthorhombic .cube files

### ISSUE THREAD

**Issue by alejandrogallo** [3dmol/3Dmol.js#167]
When loading cube files, it seems to be a problem with cells which are not orthorhombic. 
I'm uploading an example where it does not work, I have opened the file 
with the vesta program and it works fine. I'll also look into it. 

[issue_example.zip](https://github.com/3dmol/3Dmol.js/files/215040/issue_example.zip)

**Comment by dkoes** [comment]
Yes, both cube and ccp4 assume orthogaonality.  This definitely an open issue.

**Comment by alejandrogallo** [comment]
Hi, fair enough, thanks for the quick response, I think I'll look into it since I need it and if it comes out right I'll issue a pull request.

**Comment by dkoes** [comment]
That would be fantastic, especially if you could write a general function that can be applied for both cube and ccp4

**Comment by dkoes** [comment]
All I'm seeing with your example is a blank page and I don't see where the cube file is supposed to be loaded.

**Comment by alejandrogallo** [comment]
Hi, sorry, probably you have to scroll down to press the render button, I'm preparing this to leave at the directory of some calculations, and in the javascript file js/nv.js you have the variables CONTCAR and CHGCAR which at this time select which files are to be loaded, files with .cubic name on them are of course cubic, the rest are non-orthorhombic.

**Comment by dkoes** [comment]
Okay, I feel silly. I did need to scroll.  I think there may be a relatively simple fix for both cube and ccp4 so I'm looking into this now.

**Comment by alejandrogallo** [comment]
I'm also looking into it, since I am anyways interested in contributing to the code, but I have to become familiar with the structure ;)

**Comment by dkoes** [comment]
I just pushed fixes for ccp4.  Seems to work (but could use more testing):
http://3dmol.csb.pitt.edu/tests/check/test45.html
I'll look at cube now.

**Comment by dkoes** [comment]
I just pushed a commit (8a1a849) that may fix this.  I can't tell from the example if it is doing the right thing.  When I load the file in VMD I found out that VMD can't handle non-orthorhombic boxes either.  Please let me know if this is right.

**Comment by alejandrogallo** [comment]
Hi! Amazing work. I think it is just right, I get a problem with my example though. 
When I use cif files as structure input files I get a weird displacement of the volumetric data 
with respect to the cif model. However I believe this has to do with the cif parser. 

I wrote a new parser for vasp structure files (POSCARS), which I named "vasp" or "VASP" following your examples. It works fine and when I import my model using this new parser it works perfectly, so I really think it is a cif issue. Should I create a pull request for the vasp parser? I will also write a vasp parser for the volumetric data (CHG files). 
[example.zip](https://github.com/3dmol/3Dmol.js/files/216962/example.zip)
